### ISSUE TITLE
Extra whitespace on image

### ISSUE THREAD

**Issue by kylebarron** [kylebarron/stata_kernel#62]
@mcaceresb 
With the new PR, there's extra whitespace around images that are sent back to Jupyter. Compare

before:
![image](https://raw.githubusercontent.com/kylebarron/stata_kernel/master/docs/src/img/atom.png)

after:

![image](https://user-images.githubusercontent.com/15164633/43999045-c00d1cd0-9dd1-11e8-9af1-17eba45a91dd.png)

**Comment by mcaceresb** [comment]
Fixed in the `dev-magics` branch. Making a PR. Btw the error regex wasn't always working for me; I changed it to `(\r\n|$)` at the end.

I also don't get a problem with commands returning both text and images in hydrogen?
![magic_hydrogen_plot](https://user-images.githubusercontent.com/4219197/44003542-f33915f0-9e21-11e8-97c4-600086aac39d.png)

**Comment by kylebarron** [comment]
@mcaceresb is the bottom of the graph cut off in your example or is that how Stata exports it?

**Comment by mcaceresb** [comment]
Ah, it's cut off! Sadness. Back to the drawing board. Def had the right idea, tho.

**Comment by mcaceresb** [comment]
This is turning out to be very difficult to solve. I can either have the full image with the whitespace on hydrogen on have it cropped =/; still looking for a fix...

**Comment by kylebarron** [comment]
Ok. I can look into it as well if you'd like.

**Comment by mcaceresb** [comment]
I made a mistake [here](https://github.com/kylebarron/stata_kernel/blob/master/stata_kernel/stata_session.py#L733-L736). Lines 735 and 735 should be swapped. That fixes the cropping but brings back the whitespace.

This harkens back to the `_fix_svg_size` function in `stata_session.py`. SVG images don't automagically resize just because the kernel says so. Hence I fiddle with the "width" and "height" attributes in the SVG metadata. But this introduces whitespace in Hydrogen because I think Hydrogen keeps the original aspect ratio for some reason...

**Comment by mcaceresb** [comment]
Ok, so, the way to fix it is to do away with this `_fix_svg_size` function and instead just specify width and height in `graph export`.

```
        graph_export_size = {
            'pdf': None,
            'svg': ' width({0}px) height({1}px)',
            'tif': ' width({0}) height({1})',
            'png': ' width({0}) height({1})'}
        self.graph_size = graph_export_size[self.graph_format]
```

And then

```
        if self.graph_size:
            export_size = self.graph_size.format(*img_metadata.values())
        else:
            export_size = ''
```

and just add that to the export statement.

**Comment by kylebarron** [comment]
That makes a lot more sense. Thanks. Then we don't need the xml minidom stuff at all?

**Comment by mcaceresb** [comment]
No.

**Comment by mcaceresb** [comment]
I'll do it as a PR later from the magics branch. I also want to remove all the options magics I've made and put them in a single `%set` magic, which will be more familiar for Stata users.

**Comment by kylebarron** [comment]
Ok. It might be simpler to merge if it's its own PR

**Comment by mcaceresb** [comment]
Ok. I'll do it as its own PR. Can you assign me the issue?

**Comment by kylebarron** [comment]
Yeah I was originally thinking something like `%set` or `%config` where all of the settings are also values in the global `~/.stata_kernel.conf`. That way `%set` or `%config` sets them for the rest of the session but if the user wants them to be set permanently they can set them in the conf file.

https://github.com/kylebarron/stata_kernel/issues/40

**Comment by kylebarron** [comment]
![image](https://user-images.githubusercontent.com/15164633/44109581-6337c55e-9fcb-11e8-8382-431358c0569c.png)
idk why it isn't letting me assign it to you

**Comment by mcaceresb** [comment]
You can do `%set --permanently` to edit the config file.

**Comment by kylebarron** [comment]
Ok awesome.

I'm feeling exhausted and a little burnt out. I think I need to take a day or two off.

**Comment by kylebarron** [comment]
Yes, but 1) I happen to have been injured and unable to go cycling and 2) my significant other has been out of town. That's what opened up a several dozen hours in 10 days to work on this. I want to get a bit more done, at least the big async rewrite, in the next few days before those time sinks return.

**Comment by mcaceresb** [comment]
Well, both do sound like better quality time.

Hope you get better soon!
